core/src/main/java/org/eclipse/hono/util/MessageHelper.java

```diff
@@ -527,4 +557,44 @@ public static void setCreationTime(final Message msg) {
         }
     }
 
+    /**
+     * Construct a message consumer for event messages that detects an indication that a device is ready to receive an upstream
+     * message and delegates to the passed callback handlers.
+     * <p>
+     * Note that there are two different ways how this can be detected:
+     * <ul>
+     *     <li>a specific event (that has the Content-Type {@link EventConstants#CONTENT_TYPE_EMPTY_NOTIFICATION}</li>
+     *     <li>any other event that carries the AMQP 1.0 application property {@link MessageHelper#APP_PROPERTY_DEVICE_TTD}</li>
+     * </ul>
+     *
+     * @param tenantId The tenant this handler is scoped to.
+     * @param messageConsumer Message consumer that handles generic messages.
+     * @param notificationReadyToDeliverConsumer BiConsumer&lt;String,String&gt; that is called when the message indicates the current readiness
+     *                                           to receive an upstream message. Maybe {@code null}. Provided is the tenantId
+     *                                           and deviceId.
+     * @return The message consumer that invokes the specific handlers depending on the event type.
+     */
+    public static Consumer<Message> createMessageDemuxHandler(final String tenantId,
```

Comment:
IMHO this should go into a separate class (`MessageDemuxer`?) that implements `Consumer<Message>` and has this method as its static factory method.

BTW what happened to @ctron 's idea of providing a map of content-types to handlers as the param?

Reply:
Now that I understand the purpose of this, I do not think that it is a demuxer anymore. It is more of a filter or a *tap*.

Reply:
Yes, that I also like better, and it reduces the already big MessageHelper class.
Will move it to an own class.

To @ctron 's idea:
it seemed to me a good point when still thinking to only "demux" on content-types. But since this is not the case anymore, I do not think it is still matching, at least not currently.
So in this PR I decided to let the signature as it was - passing in explicit callback methods.